frontend/app/controllers/Event.scala

```diff
@@ -38,10 +40,19 @@ trait Event extends Controller {
   private def BuyAction(id: String) = NoCacheAction andThen recordBuyIntention(id) andThen
     authenticated(onUnauthenticated = notYetAMemberOn(_)) andThen memberRefiner()
 
-  def details(id: String) = CachedAction { implicit request =>
-    EventbriteService.getEvent(id).map { event =>
-      eventDetail(event, request.path)
-    }.getOrElse(Redirect(Config.guardianMembershipUrl))
+  def details(slug: String) = NoCacheAction { implicit request =>
+    val eventOpt = for {
+      id <- EBEvent.slugToId(slug)
```

Comment:
This is more of a learning question than a review, but does redefining `id` here only apply to the scope of the `for`?